src/web/toga_web/window.py

```diff
@@ -54,25 +47,20 @@ def clear_content(self):
                 child._impl.container = None
 
     def set_content(self, widget):
-        self.interface.factory.not_implemented('Window.set_content()')
         widget.viewport = WebViewport()
+
         # Add all children to the content widget.
-        for child in widget.interface.children:
-            child._impl.container = widget
+        # for child in widget.interface.children:
+        #     # child._impl.container = widget
+        self.native.appendChild(widget.native)
 
     def show(self):
-        self.interface.factory.not_implemented('Window.show()')
-        # self.native.show_all()
-
-        # # Now that the content is visible, we can do our initial hinting,
-        # # and use that as the basis for setting the minimum window size.
-        # self.interface.content._impl.rehint()
-        # self.interface.content.style.layout(self.interface.content, Viewport(0, 0))
-        # self.interface.content._impl.min_width = self.interface.content.layout.width
-        # self.interface.content._impl.min_height = self.interface.content.layout.height
+        app_placeholder = js.document.getElementById("app-placeholder")
+        app_placeholder.appendChild(self.native)
```

Comment:
This should be a no-op if called a second time.

Reply:
True; there were a bunch of other edge cases related to setting main window content as well. Testing these is difficult without more widget coverage, but I've updated the implementation so that it still works, but points in the general direction of what a general solution should look like.